EIPS/eip-00.md

```diff
@@ -0,0 +1,790 @@
+initialized with contract code.
+
+#### Sender validation frame
+
+We define the following Solidity struct to represent the AA transaction on-chain:
+
+```solidity
+
+struct TransactionType4 {
+    address sender;
+    uint256 nonce;
+    uint256 validationGasLimit;
+    uint256 paymasterGasLimit;
+    uint256 callGasLimit;
+    uint256 maxFeePerGas;
+    uint256 maxPriorityFeePerGas;
+    uint256 builderFee;
+    bytes paymasterData;
+    bytes deployerData;
+    bytes callData;
+    bytes signature;
+}
+
+```
+
+We then define the following Solidity method and the `sender` of the transaction is invoked with the corresponding data:
+
+```solidity
+
+function validateTransaction(uint256 version, bytes32 txHash, bytes transaction) external returns (uint256 validationData);
+
+```
+
+The gas limit of this frame is set to `validationGasLimit - senderCreationGasUsed`.
+The `transaction` parameter is interpreted as an ABI encoding of `TransactionType4`.
+The `version` parameter is added in order to maintain the Solidity method ID in case of changes to this struct
+in future revisions of this EIP.
+
+The amount of gas used by this frame is referred to as `senderValidationGasUsed`.
+
+The frame must return 32 bytes `validationData` that is interpreted as:
+
+```solidity
+
+abi.encodePacked(MAGIC_VALUE_SENDER, validBefore, validAfter)
+
+```
+
+In order to allow a gas estimation to determine the amount of gas that this frame
+requires to complete successfully while not having the actual `signature` value, this function
+should avoid reverting on invalid signature, and should return a value different from `MAGIC_VALUE_SENDER`.
+
+The `validateTransaction` function can choose to revert on any condition that can be satisfied during gas estimation.
+
+#### Paymaster validation frame
+
+The `paymaster` of the transaction, if specified, is invoked with the following data:
+
+```solidity
+
+function validatePaymasterTransaction(uint256 version, bytes32 txHash, bytes transaction) external returns (bytes context, uint256 validationData);
+
+```
+
+The gas limit of this frame is set to `paymasterGasLimit`.
+
+The amount of gas used by this frame is referred to as `paymasterValidationGasUsed`.
+
+The `transaction` parameter is interpreted as an ABI encoding of `TransactionType4`.
+
+The frame must return a bytes array that is interpreted as:
+
+```solidity
+
+abi.encode(context, MAGIC_VALUE_PAYMASTER, validBefore, validAfter)
+
+```
+
+Same as in the [`sender` validation frame](#sender-validation-frame), in order to support gas estimation this
+frame should return a value different from `MAGIC_VALUE_PAYMASTER` for conditions that cannot be satisfied
+before signing.
+
+The size of the `context` byte array may not exceed `MAX_CONTEXT_SIZE` for a transaction to be considered valid.
+
+#### Sender execution frame
+
+The `sender` address is invoked with `callData` input.
+
+The gas limit of this frame is set to `callGasLimit`.
```

Comment:
Yes, in the pseudocode it's called with `callGasLimit - calldataCost` (line 570).  @forshtat I think the correct one is the one in the pseudocode rather than this one, isn't it?